rasterio/crs.pyx

```diff
@@ -924,27 +924,19 @@ cdef class CRS:
         cdef OGRSpatialReferenceH osr_s = NULL
         cdef OGRSpatialReferenceH osr_o = NULL
         cdef CRS crs_o
-
+        cdef const char* options[2]
+        options[0] = b"IGNORE_DATA_AXIS_TO_SRS_AXIS_MAPPING=YES"
```

Comment:
@AndrewAnnex @snowman2 will this result in `EPSG:4326` and `OGC:CRS84` being evaluated as equal? Do we want that? I think we should have a new test or two for this case.

Reply:
The main thing for rasterio to decide is whether or not axis order needs to be taken into account for whether the CRS is equal. For pyproj, it defaults to the default PROJ provides as axis order matters when doing transformations.

For pyproj, these are the same:
```
>>> from pyproj import CRS
>>> CRS(init="EPSG:4326") == CRS("OGC:CRS84")
True
```
That makes sense to since they have the same CRS parameters and axis order.

These CRS are different because the have a different axis order:
```python
>>> CRS("EPSG:4326") == CRS("OGC:CRS84")
False
```
But, they can ignore axis order in the comparison if they want to:
```python
>>> CRS("EPSG:4326").equals(CRS("OGC:CRS84"), ignore_axis_order=True)
True
```

However, in rasterio, axis order is ignored entirely when doing transformations. So, ignoring the axis order when comparing the CRS, as is done in this PR, makes sense to me.

Reply:
@snowman2 thanks for the explainer as this was a general question I had as well. Currently this PR doesn't break from the existing behavior, as I believe CRS84 gets converted to the 4326 code with the `to_epsg` method. But it could also make sense to add a .equals method to rasterios' CRS to enable more flexibility like in pyproj

Reply:
@sgillies do you still want other tests? `test_is_same_crs`, `test_crs_OSR_equivalence` and `test_crs_OSR_no_equivalence` seem to cover things

Reply:
@AndrewAnnex yes, I'd like to see a test demonstrating that the following (rasterio 1.4.1) is preserved

```
>>> CRS.from_epsg(4326) == CRS.from_string("OGC:CRS84")
False
```

Reply:
@sgillies I believe that they should be equal with this pr (running the test suite locally now) as the only differences are the axis order and the authority. 

Why that was unequal before is unclear to me, maybe the authority isn't ignored by default?

Reply:
@sgillies they are equal CRSs with the state of the PR currently, I think though this is the behavior @snowman2 wants. In rasterio 1.4.1 likely they are not equal because the axis order is not ignored in the CRS test (this PR changes that to do so). I think setting `IGNORE_DATA_AXIS_TO_SRS_AXIS_MAPPING=NO` (the default) causes other tests to fail where the EPSG codes used to resolve to the same code, but would have otherwise failed at the fallback IsSame check.

Reply:
maybe we could have a `CRS.equals()`  method (as in pyproj) which default with `ignore_axis_order=false`, and reuse it in `__eq__`, with would preserve the behaviour from 1.4 bet let users to either use ` crs == crs2` or `crs.equals(crs2, ignore_axis_order=False|True)`  

https://github.com/pyproj4/pyproj/blob/39cf3c95c7b81e364fbff275a62fd44ba8113cff/pyproj/_crs.pyx#L3162-L3179

Reply:
@vincentsarago yeah I agree that'd be a good idea and I am happy to add it + some docs for this in this PR, or as a separate PR. a +1 from @sgillies  though would be great to hear before I go about implementing that

Reply:
Sure, I think that parity and agreement with pyproj would be good for 1.5.0.

For 1.4.x, I welcome better performance, but don't welcome a change of behavior. That's why I'm concerned about `b"IGNORE_DATA_AXIS_TO_SRS_AXIS_MAPPING=YES"`.